packages/contracts-bedrock/scripts/Deploy.s.sol

```diff
@@ -62,14 +64,30 @@ contract Deploy is Deployer {
 
         string memory path = string.concat(vm.projectRoot(), "/deploy-config/", deploymentContext, ".json");
         cfg = new DeployConfig(path);
-
         console.log("Deploying from %s", deployScript);
         console.log("Deployment context: %s", deploymentContext);
     }
 
     /// @notice Deploy all of the L1 contracts
     function run() public {
-        console.log("Deploying L1 system");
+        console.log("Deploying a fresh OP Stack including SuperchainConfig");
+        address supConf = setupSuperchainConfig();
+        setupOpChain(supConf);
+    }
+
+    /// @notice Deploy a full system with a new SuperchainConfig
+    function setupSuperchainConfig() public returns (address supConf_) {
+        supConf_ = deploySuperchainConfigProxy();
+        deploySuperchainConfig();
+        initializeSuperchainConfig();
+    }
+
+    /// @notice Deploy a new OP Chain, with an existing SuperchainConfig provided
+    function setupOpChain(address supConf) public {
```

Comment:
Technically setting up an op chain doesn't need to deploy the implementations

Reply:
We don't need to worry about this now and can eventually follow up with a deploy script that has clean abstractions for:
- creating superchain (deploy superchain config)
- deploying implementations
- deploying proxies
- initializing proxies

Reply:
Good call out. I think that likely will need to be resolved before merging back into `develop`. Going to leave this comment open for now.

Reply:
Resolving now.